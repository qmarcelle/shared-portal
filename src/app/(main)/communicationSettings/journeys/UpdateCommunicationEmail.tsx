import { ChangeAuthDeviceSlide } from '@/components/composite/ChangeAuthDeviceSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import { AppLink } from '@/components/foundation/AppLink';
import {
    ModalChildProps,
    useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { TextField } from '@/components/foundation/TextField';
import { useState } from 'react';
import { InputModalSlide } from '../../../../components/composite/InputModalSlide';
import { Spacer } from '../../../../components/foundation/Spacer';
import { TextBox } from '../../../../components/foundation/TextBox';

const headerText = 'Confirm Email Address';
interface UpdateCommunicationEmailProps {
  email: string;
}

export const UpdateCommunicationEmail = ({
  changePage,
  pageIndex,
  email,
}: ModalChildProps & UpdateCommunicationEmailProps) => {
  const { dismissModal } = useAppModalStore();
  /* const initChange = () => {
    changePage!(1, true);
  };
  function changePageIndex(index: number, showback = true) {
    changePage?.(index, showback);
  } */
  const [mainAuthDevice, setMainAuthDevice] = useState(email);
  const [newAuthDevice, setNewAuthDevice] = useState('');
  const [confirmCode, setConfirmCode] = useState('');

  const initNewDevice = async () => {
    // Do API call for new device
    setMainAuthDevice(newAuthDevice);
    changePage?.(1, true);
  };

  const submitCode = async () => {
    // Do API call for submit code
    changePage?.(2, false);
  };

  const pages = [
    <ChangeAuthDeviceSlide
      key={0}
      label="Update Email Address"
      subLabel="Enter the new email address you'd like to use for communications and security settings."
      actionArea={
        <TextField
          valueCallback={(val) => setNewAuthDevice(val)}
          label="Email Address"
        />
      }
      cancelCallback={() => dismissModal()}
      nextCallback={
        newAuthDevice.length > 5 ? () => initNewDevice() : undefined
      }
    />,
    <InputModalSlide
      key={1}
      label={headerText}
      subLabel="We just need to verify your email. Enter the security code we sent to:"
      actionArea={
        <Column>
          <TextBox
            className="body-1 text-center font-bold"
            text={mainAuthDevice}
          />
          <Spacer size={32} />
          <TextField
            type="text"
            valueCallback={(val) => setConfirmCode(val)}
            label="Enter Security Code"
          ></TextField>
          <Spacer size={16} />
          <AppLink className="self-start" label="Resend Code" />
          <Spacer size={32} />
        </Column>
      }
      cancelCallback={() => dismissModal()}
      nextCallback={confirmCode.length > 5 ? () => submitCode() : undefined}
    />,

    <SuccessSlide
      key={2}
      label="Email Address Updated"
      body={
        <Column className="items-center">
          <TextBox className="text-center" text="Your email address is:" />
          <Spacer size={16} />
          <TextBox text={mainAuthDevice} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
