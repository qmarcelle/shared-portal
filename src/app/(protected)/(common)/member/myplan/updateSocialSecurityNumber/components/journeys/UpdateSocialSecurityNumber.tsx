import { ErrorDisplaySlide } from '@/components/composite/ErrorDisplaySlide';
import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { TextField } from '@/components/foundation/TextField';
import { useEffect, useRef, useState } from 'react';
import { updateSSNData } from '../../actions/updateSSNData';
import { UpdateSSNRequest } from '../../models/app/updateSSNRequest';

interface UpdateSocialSecurityNumberJourneyProps {
  memberName: string;
  successCallback: () => void;
}

const headerText = 'Update or Add a Social Security Number';

export const UpdateSocialSecurityNumberJourney = ({
  changePage,
  pageIndex,
  memberName,
  successCallback,
}: ModalChildProps & UpdateSocialSecurityNumberJourneyProps) => {
  const { dismissModal } = useAppModalStore();
  const ssnInputRef = useRef<HTMLInputElement>(null);
  const [securityNumber, setSecurityNumber] = useState('***-**-****');

  useEffect(() => {
    if (pageIndex === 0) {
      // Focus SSN input on first page
      ssnInputRef.current?.focus();
    }
  }, [pageIndex]);

  const addSecurityNumber = (value: string) => {
    if (!isNaN(Number(value))) {
      setSecurityNumber(value);
    }
  };

  function changePageIndex(index: number, showback = true) {
    changePage?.(index, showback);
  }

  const submitSSNCode = async () => {
    try {
      const request: UpdateSSNRequest = {
        ssn: securityNumber,
      };
      const response = await updateSSNData(request);
      if (response.data?.message === 'Updated SSN') {
        successCallback();
        changePageIndex?.(1, false);
      }
    } catch (errorMessage: unknown) {
      changePageIndex?.(2, false);
    }
  };

  const pages = [
    <InputModalSlide
      key={0}
      label={headerText}
      subLabel="Enter the social security number for:"
      actionArea={
        <Column>
          <TextBox
            className="font-bold text-center"
            text={memberName}
            ariaLabel={`Enter SSN for ${memberName}`}
          />
          <Spacer size={32} />
          <TextField
            ref={ssnInputRef}
            hint="***-**-****"
            valueCallback={(val) => addSecurityNumber(val)}
            maxLength={9}
            label="Social Security Number"
            ariaLabel="Enter your social security number"
            ariaDescribedBy="ssn-description"
            type="password"
          />
          <TextBox
            id="ssn-description"
            className="sr-only"
            text="Enter your 9-digit social security number. The number will be masked for security."
          />
          <Spacer size={32} />
        </Column>
      }
      nextCallback={
        securityNumber.length > 5 ? () => submitSSNCode() : undefined
      }
      cancelCallback={() => dismissModal()}
      buttonLabel="Save Changes"
    />,

    <SuccessSlide
      key={1}
      label="Social Security Number Updated"
      body={
        <Column className="items-center">
          <TextBox
            className="text-center"
            text="Your social security number has been successfully updated."
            ariaLabel="Your social security number has been successfully updated."
          />
          <Spacer size={32} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
    <ErrorDisplaySlide
      key={2}
      label="Something went wrong."
      body={
        <Column className="items-center">
          <TextBox
            className="text-center"
            text="We're unable to update your information at this time. Please try again later."
            ariaLabel="We're unable to update your information at this time. Please try again later."
          />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
