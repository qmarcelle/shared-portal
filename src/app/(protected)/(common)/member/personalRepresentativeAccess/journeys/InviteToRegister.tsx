import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { TextField } from '@/components/foundation/TextField';

interface InviteToRegisterProps {
  memberName: string;
}

export const InviteToRegister = ({
  changePage,
  pageIndex,
  memberName,
}: ModalChildProps & InviteToRegisterProps) => {
  const { dismissModal } = useAppModalStore();
  const initChange = () => {
    changePage!(1, false);
  };

  const pages = [
    <InputModalSlide
      key={0}
      label="Invite to Register"
      subLabel="You're inviting:"
      buttonLabel="Send Invite"
      actionArea={
        <Column>
          <TextBox className="body-1 text-center font-bold" text={memberName} />
          <Spacer size={32} />
          <TextBox
            className="text-center"
            text="We'll send an email inviting them to register an account. We just need their email."
          />
          <Spacer size={32} />
          <TextField type="text" label="Their Email Address"></TextField>
          <Spacer size={32} />
          <TextField
            type="text"
            label="Confirm Their Email Address"
          ></TextField>
          <Spacer size={32} />
        </Column>
      }
      cancelCallback={() => dismissModal()}
      nextCallback={initChange}
    />,
    <SuccessSlide
      key={1}
      label="Invitation to Register Sent"
      body={
        <Column className="items-center ">
          <TextBox
            className="text-center"
            text="We've sent an email inviting this member to register. Once they've completed registration, you'll be able to request access to their information"
          />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
