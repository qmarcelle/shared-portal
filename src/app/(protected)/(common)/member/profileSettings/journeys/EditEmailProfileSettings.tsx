import { ChangeAuthDeviceSlide } from '@/components/composite/ChangeAuthDeviceSlide';
import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import { AppLink } from '@/components/foundation/AppLink';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { TextField } from '@/components/foundation/TextField';
import { useEffect, useRef, useState } from 'react';

interface EditEmailSettingsJourneyProps {
  email: string;
}

export const EditEmailProfileSettings = ({
  changePage,
  pageIndex,
  email,
}: ModalChildProps & EditEmailSettingsJourneyProps) => {
  const { dismissModal } = useAppModalStore();
  const emailInputRef = useRef<HTMLInputElement>(null);
  const verificationInputRef = useRef<HTMLInputElement>(null);

  const [mainAuthDevice, setMainAuthDevice] = useState(email);
  const [newAuthDevice, setNewAuthDevice] = useState('');
  const [confirmCode, setConfirmCode] = useState('');

  useEffect(() => {
    if (pageIndex === 0) {
      // Focus email input on first page
      emailInputRef.current?.focus();
    } else if (pageIndex === 1) {
      // Focus verification code input on verification page
      verificationInputRef.current?.focus();
    }
  }, [pageIndex]);

  const initNewDevice = async () => {
    // Do API call for new device
    setMainAuthDevice(newAuthDevice);
    changePage?.(1, true);
  };

  const submitCode = async () => {
    // Do API call for submit code
    changePage?.(2, false);
  };

  const pages = [
    <ChangeAuthDeviceSlide
      key={0}
      label="Update Email Address"
      subLabel="Enter the new email address you'd like to use for communications and security settings."
      actionArea={
        <TextField
          ref={emailInputRef}
          valueCallback={(val) => setNewAuthDevice(val)}
          label="Email Address"
          ariaLabel="Enter your new email address"
          ariaDescribedBy="email-description"
          type="email"
        />
      }
      cancelCallback={() => dismissModal()}
      nextCallback={
        newAuthDevice.length > 5 ? () => initNewDevice() : undefined
      }
    />,
    <InputModalSlide
      key={1}
      label="Confirm Email Address"
      subLabel="We just need to verify your email. Enter the security code we sent to:"
      actionArea={
        <Column>
          <Spacer size={16} />
          <TextBox
            className="font-bold text-center"
            text={mainAuthDevice}
            ariaLabel="Your email address"
          />
          <Spacer size={32} />
          <TextField
            ref={verificationInputRef}
            type="text"
            valueCallback={(val) => setConfirmCode(val)}
            label="Enter Security Code"
            ariaLabel="Enter the verification code sent to your email"
            ariaDescribedBy="verification-code-description"
          />
          <TextBox
            id="verification-code-description"
            className="sr-only"
            text="Enter the 6-digit verification code sent to your email address"
          />
          <Spacer size={16} />
          <AppLink
            className="self-start !p-0"
            label="Resend Code"
            ariaLabel="Click to resend the verification code"
          />
          <Spacer size={32} />
        </Column>
      }
      cancelCallback={() => dismissModal()}
      nextCallback={confirmCode.length > 5 ? () => submitCode() : undefined}
    />,
    <SuccessSlide
      key={2}
      label="Email Address Updated"
      body={
        <Column>
          <TextBox
            className="text-center"
            text="Your email address is: "
            ariaLabel="Your updated email address is"
          />
          <Spacer size={16} />
          <TextBox
            className="text-center font-bold"
            text={mainAuthDevice}
            ariaLabel={mainAuthDevice}
          />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
