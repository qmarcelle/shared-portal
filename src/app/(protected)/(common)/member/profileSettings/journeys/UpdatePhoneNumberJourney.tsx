import { InitModalSlide } from '@/components/composite/InitModalSlide';
import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import { AppLink } from '@/components/foundation/AppLink';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Radio } from '@/components/foundation/Radio';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { TextField } from '@/components/foundation/TextField';
import { useEffect, useRef } from 'react';

const bottomNote =
  'By sending the code I agree to receive a one-time security code. Message and data rates may apply, Subject to terms and condition.';

/* eslint-disable */

interface InviteToRegisterProps {
  phoneNumber: string;
}

export const UpdatePhoneNumberJourney = ({
  changePage,
  pageIndex,
  phoneNumber,
}: ModalChildProps & InviteToRegisterProps) => {
  const { dismissModal } = useAppModalStore();
  const phoneInputRef = useRef<HTMLInputElement>(null);
  const verificationInputRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    if (pageIndex === 0) {
      // Focus phone input on first page
      phoneInputRef.current?.focus();
    } else if (pageIndex === 2) {
      // Focus verification code input on verification page
      verificationInputRef.current?.focus();
    }
  }, [pageIndex]);

  const pages = [
    <InputModalSlide
      key={0}
      label="Update Phone Number"
      subLabel="Enter the new phone number you'd like to use for communications and security settings."
      buttonLabel="Next"
      actionArea={
        <Column className="items-center">
          <Spacer size={32} />
          <TextField
            ref={phoneInputRef}
            label="Phone Number"
            ariaLabel="Enter your new phone number"
            ariaDescribedBy="phone-number-description"
          />
          <TextBox
            id="phone-number-description"
            className="sr-only"
            text="Enter your new phone number in the format (XXX) XXX-XXXX"
          />
          <Spacer size={24} />
        </Column>
      }
      nextCallback={() => changePage?.(1, true)}
      cancelCallback={() => dismissModal()}
    />,
    <InitModalSlide
      key={1}
      label="Confirm Phone Number"
      subLabel={
        <Column>
          <TextBox
            className="text-center"
            text="We just need to verify your phone number. How would you like to receive the security code?"
          />
          <Spacer size={24} />
          <Column>
            <Radio
              label="Text a code to (123) 456-0000"
              selected={true}
              ariaLabel="Receive verification code via text message"
            />
            <Radio
              label="Call with a code to (123) 456-0000"
              selected={false}
              ariaLabel="Receive verification code via phone call"
            />
          </Column>
        </Column>
      }
      changeAuthButton={undefined}
      bottomNote={<TextBox className="body-2" text={bottomNote} />}
      buttonLabel="Send Code"
      nextCallback={() => changePage?.(2, true)}
      cancelCallback={() => dismissModal()}
    />,
    <InputModalSlide
      key={2}
      label="Confirm Phone Number"
      subLabel="Enter the security code we sent to:"
      actionArea={
        <Column className="items-center">
          <TextBox className="font-bold" text={phoneNumber} />
          <Spacer size={32} />
          <TextField
            ref={verificationInputRef}
            label="Enter Security Code"
            ariaLabel="Enter the verification code sent to your phone"
            ariaDescribedBy="verification-code-description"
          />
          <TextBox
            id="verification-code-description"
            className="sr-only"
            text="Enter the 6-digit verification code sent to your phone number"
          />
          <AppLink
            className="self-start"
            label="Resend Code"
            ariaLabel="Click to resend the verification code"
          />
          <Spacer size={32} />
        </Column>
      }
      nextCallback={() => changePage?.(3, true)}
      cancelCallback={() => dismissModal()}
    />,
    <SuccessSlide
      key={3}
      label="Phone Number Updated"
      body={
        <Column className="items-center">
          <TextBox className="text-center" text="Your phone number is:" />
          <Spacer size={16} />
          <TextBox className="font-bold" text={phoneNumber} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
