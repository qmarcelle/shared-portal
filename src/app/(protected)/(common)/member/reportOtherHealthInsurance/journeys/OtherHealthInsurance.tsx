import { InitModalSlide } from '@/components/composite/InitModalSlide';
import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Radio } from '@/components/foundation/Radio';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { useRef, useState } from 'react';

import { AddMemberDetails } from '@/models/add_member_details';
import AddMemberPlan from '@/components/AddMemberPlan';
import OtherHealthInsurancePlan from '@/components/OtherHealthInsurancePlan';
import SelectMemberPlan from '@/components/SelectMemberPlan';
export type OtherHealthInsuranceProps = {
  memberDetails: AddMemberDetails[];
};
export const OtherHealthInsurance = ({
  changePage,
  pageIndex,
  memberDetails,
}: ModalChildProps & OtherHealthInsuranceProps) => {
  const [selectedData, setSelectedData] = useState(false);
  const [selectedCheckbox, setSelectedCheckbox] = useState<string[] | null>([]);
  const checkCountRef = useRef<number>(0);
  const [checkboxState, setCheckboxState] = useState({
    medicalPlan: false,
    dentalPlan: false,
    medicarePlan: false,
  });

  const [isAnyChecked, setIsAnyChecked] = useState(false);

  const handleCheckboxChange = (checkboxValue: string[]) => {
    const newState = { ...checkboxState };
    // Toggle checkbox state
    if (checkboxValue.includes('medicarePlan')) {
      newState.medicarePlan = !newState.medicarePlan;
      if (newState.medicarePlan) {
        newState.medicalPlan = false;
        newState.dentalPlan = false;
      }
    } else if (checkboxValue.includes('medicalPlan')) {
      newState.medicalPlan = !newState.medicalPlan;
      if (newState.medicalPlan) {
        newState.medicarePlan = false;
      }
    } else if (checkboxValue.includes('dentalPlan')) {
      newState.dentalPlan = !newState.dentalPlan;
      if (newState.dentalPlan) {
        newState.medicarePlan = false;
      }
    }

    // Update checkbox states
    setCheckboxState(newState);

    // Set the selected checkbox value
    setSelectedCheckbox(
      newState.medicarePlan
        ? ['medicarePlan']
        : newState.medicalPlan
          ? ['medicalPlan']
          : newState.dentalPlan
            ? ['dentalPlan']
            : null,
    );

    // Check if any checkbox is selected
    setIsAnyChecked(
      newState.medicarePlan || newState.medicalPlan || newState.dentalPlan,
    );
  };

  const handleNextCall = () => {
    checkCountRef.current += 1;
    if (
      checkboxState.medicalPlan &&
      checkboxState.dentalPlan &&
      checkCountRef.current === 1
    ) {
      changePage?.(2, true);
      setSelectedCheckbox(['dentalPlan']);
    } else {
      changePage?.(4, true);
    }
  };

  function handleClick() {
    setSelectedData(true);
    setSelectedData(!selectedData);
  }

  const { dismissModal } = useAppModalStore();

  const pages = [
    <InitModalSlide
      key="first"
      label="Other Health Insurance"
      subLabel={
        <Column>
          <TextBox
            className="text-center"
            text="Does this plan member have other insurance?"
          />
          <Spacer size={24} />

          <Column>
            <Radio selected={!selectedData} callback={handleClick} label="No" />
            <Spacer size={8} />
            <Radio label="Yes" selected={selectedData} callback={handleClick} />
          </Column>
        </Column>
      }
      changeAuthButton={undefined}
      buttonLabel="Next"
      nextCallback={() => changePage?.(1, true)}
      cancelCallback={() => dismissModal()}
    />,
    <InputModalSlide
      key="second"
      label="Other Health Insurance"
      subLabel=""
      actionArea={
        <OtherHealthInsurancePlan
          checkboxState={checkboxState}
          onCheckboxChange={handleCheckboxChange}
          selectedData={selectedData}
        />
      }
      buttonLabel={selectedData ? 'Next' : 'Save Answer'}
      nextCallback={
        selectedData
          ? isAnyChecked
            ? () => changePage?.(2, true)
            : undefined
          : () => changePage?.(4, true)
      }
      cancelCallback={() => dismissModal()}
    />,
    <InputModalSlide
      key="third"
      label=""
      subLabel=""
      actionArea={
        <AddMemberPlan
          selectedCheckbox={selectedCheckbox}
          memberDetails={memberDetails}
        />
      }
      buttonLabel="Next"
      nextCallback={() => changePage?.(3, true)}
      cancelCallback={() => dismissModal()}
    />,
    <InputModalSlide
      key="Fourth"
      label=""
      subLabel=""
      actionArea={<SelectMemberPlan selectedCheckbox={selectedCheckbox} />}
      buttonLabel="Save Answer"
      nextCallback={handleNextCall}
      cancelCallback={() => dismissModal()}
    />,
    <SuccessSlide
      key="Fifth"
      label="Other Insurance Updated"
      body={
        <Column className="items-center">
          <TextBox
            className="text-center w-[125%]"
            text="Thanks for confirming your plan information. We'll check in with you next year. If anything changes before then, you can make updates anytime."
          />
          <Spacer size={16} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
