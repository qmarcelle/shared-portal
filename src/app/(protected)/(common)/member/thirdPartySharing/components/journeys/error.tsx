'use client';

import { useEffect } from 'react';

/**
 * Error component for thirdPartySharing/components/journeys
 * Handles errors that occur during page rendering
 */
export default function Error({
  error,
  reset,
}: {
  error: Error & { digest?: string };
  reset: () => void;
}) {
  useEffect(() => {
    console.error(error);
  }, [error]);

  return (
    <div className="flex flex-col items-center justify-center p-4">
      <div className="text-center">
        <h1 className="mb-4 text-2xl font-bold">Something went wrong!</h1>
        <p className="mb-4 text-gray-600">{error.message}</p>
        <button
          onClick={() => reset()}
          className="rounded bg-blue-500 px-4 py-2 text-white hover:bg-blue-600"
        >
          Try again
        </button>
      </div>
    </div>
  );
}