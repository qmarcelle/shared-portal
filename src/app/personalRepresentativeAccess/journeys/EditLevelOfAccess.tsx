import { InitModalSlide } from '@/components/composite/InitModalSlide';
import { InputModalSlide } from '@/components/composite/InputModalSlide';
import { SuccessSlide } from '@/components/composite/SuccessSlide';
import {
  ModalChildProps,
  useAppModalStore,
} from '@/components/foundation/AppModal';
import { Column } from '@/components/foundation/Column';
import { Radio } from '@/components/foundation/Radio';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { useState } from 'react';
import FullAndBasicAccessOption from '../components/FullAndBasicAccessOption';

const bottomNote =
  'By sending the code I agree to receive a one-time security code. Message and data rates may apply, Subject to terms and condition.';

interface InviteToRegisterProps {
  memberName: string;
}

export const EditLevelOfAccess = ({
  changePage,
  pageIndex,
  memberName,
}: ModalChildProps & InviteToRegisterProps) => {
  const [selectedData, setSelectedData] = useState(false);

  function handleClick() {
    setSelectedData(true);
    setSelectedData(!selectedData);
  }

  const { dismissModal } = useAppModalStore();

  const pages = [
    <InitModalSlide
      key="first"
      label="Edit Level Of Access"
      subLabel={
        <Column>
          <TextBox className="text-center" text="You're changing access for:" />
          <Spacer size={24} />
          <TextBox className="text-center" text="Chris Hall" />
          <Spacer size={24} />
          <Column>
            <Radio
              label="Full Access"
              subLabel="Your Representative will have access to all documents and claims, even those with sensitive information"
              selected={!selectedData}
              callback={handleClick}
            />
            <Radio
              label="Basic Access"
              subLabel="Your Representative will have access to all documents and claims, but will not be able to view sensitive information"
              selected={selectedData}
              callback={handleClick}
            />
          </Column>
        </Column>
      }
      changeAuthButton={undefined}
      buttonLabel="Next"
      nextCallback={() => changePage?.(1, true)}
      bottomNote={<TextBox className="body-2" text={bottomNote} />}
      cancelCallback={() => dismissModal()}
    />,
    <InputModalSlide
      key="second"
      label="Edit Level Of Access"
      subLabel=""
      actionArea={<FullAndBasicAccessOption selectedData={selectedData} />}
      buttonLabel="Save Permissions"
      nextCallback={() => changePage?.(2, true)}
      cancelCallback={() => dismissModal()}
    />,
    <SuccessSlide
      key="third"
      label="Level Of Access Saved"
      body={
        <Column className="items-center">
          <TextBox
            className="text-center"
            text="The Level of access to your account information has been updated for:"
          />
          <Spacer size={16} />
          <TextBox className="font-bold" text={memberName} />
        </Column>
      }
      doneCallBack={() => dismissModal()}
    />,
  ];

  return pages[pageIndex!];
};
