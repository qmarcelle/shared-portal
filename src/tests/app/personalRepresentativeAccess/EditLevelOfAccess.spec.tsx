import { EditLevelOfAccess } from '@/app/personalRepresentativeAccess/journeys/EditLevelOfAccess';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

describe('Edit Level Of Access Component', () => {
  it('should render the UI correctly', () => {
    const component = render(
      <EditLevelOfAccess
        memberName="Chris"
        pageIndex={0}
        currentAccessType="full"
        isMaturedMinor
      />,
    );
    expect(screen.getByText('Edit Level Of Access')).toBeInTheDocument();
    expect(screen.getByText('Full Access +')).toBeInTheDocument();
    expect(
      screen.getByText(
        'Your personal representative will have access to all documents and claims, but will not be able to view sensitive information.',
      ),
    ).toBeInTheDocument();
    expect(component.baseElement).toMatchSnapshot();
  });
});
