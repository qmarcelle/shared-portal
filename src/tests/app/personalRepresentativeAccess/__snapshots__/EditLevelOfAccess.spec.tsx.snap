// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Level Of Access Component should render the UI correctly 1`] = `
<body>
  <div>
    <div
      class="flex flex-col items-center"
    >
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
      <h2
        class="title-2"
      >
        Edit Level Of Access
      </h2>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
      />
      <div
        class="flex flex-col items-center  w-[358px]"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="body-1 text-center"
            style="display: block; max-width: 100%;"
          >
            Youâ€™re changing access for:
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <p
            class="body-1 text-center font-bold"
            style="display: block; max-width: 100%;"
          >
            Chris
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <div
            class="flex flex-col"
          >
            <div
              class="flex flex-row gap-1 "
            >
              <label>
                <input
                  aria-label="Full Access +"
                  id=""
                  name=""
                  type="radio"
                />
              </label>
              <div
                class="flex-col m-1"
              >
                <p
                  class="font-bold"
                >
                  Full Access +
                </p>
                <p>
                  Your personal representative and anyone they choose will have access to all documents and claims, even those with sensitive information.
                </p>
              </div>
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
            />
            <div
              class="flex flex-row gap-1 "
            >
              <label>
                <input
                  aria-label="Basic Access"
                  id=""
                  name=""
                  type="radio"
                />
              </label>
              <div
                class="flex-col m-1"
              >
                <p
                  class="font-bold"
                >
                  Basic Access
                </p>
                <p>
                  Your personal representative will have access to all documents and claims, but will not be able to view sensitive information.
                </p>
              </div>
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
            />
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <div
          class="mb-8"
        >
          <p
            class="body-1 body-2"
            style="display: block; max-width: 100%;"
          >
            Disclaimer: BlueCross BlueShield of Tennessee is not responsible for your personal representative or any third parties authorized by you or your personal representative to hae access to your health information. BlueCross BlueShield of Tennessee does not warrant that the information provided will be accurate, timely or complete.
          </p>
        </div>
        <button
          aria-label="Next"
          class="button-text primary text-center flex flex-row justify-center items-center min-w-fit null font-bold primary-color"
          type="button"
        >
          Next
        </button>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <button
          class="flex flex-row link-container undefined"
          style="max-width: max-content; height: auto; display: block;"
          type="button"
        >
          <p
            class="link !no-underline"
            style="display: block;"
          >
            Cancel
          </p>
        </button>
      </div>
    </div>
  </div>
</body>
`;
